core/src/main/java/brooklyn/entity/basic/EntityPredicates.java

```diff
@@ -271,9 +274,38 @@ public boolean apply(@Nullable Entity input) {
     // ---------------------------
 
     /**
+     * @param typeRegex a regular expression
+     * @return true if any of the interfaces implemented by the entity (including those derived) match typeRegex.
+     */
+    public static Predicate<Entity> hasInterfaceMatching(String typeRegex) {
+        return new ImplementsInterface(typeRegex);
+    }
+
+    protected static class ImplementsInterface implements SerializablePredicate<Entity> {
+        protected final Pattern pattern;
+
+        public ImplementsInterface(String typeRegex) {
+            this.pattern = Pattern.compile(typeRegex);
+        }
+
+        @Override
+        public boolean apply(@Nullable Entity input) {
+            if (input == null) return false;
+            for (Class<?> cls : ClassUtils.getAllInterfaces(input.getClass())) {
```

Comment:
We also have `brooklyn.util.javalang.Reflections.getAllInterfaces(Class)`, which does the same as in `ClassUtils`.

If we're going to keep the dependency on apache-commons lang3, then we should probably delete that brooklyn code! (Or more specifically deprecate it, and have its impl delegate to the lang3 version).

But that's not for this PR.